core/templates/pages/topics-and-skills-dashboard-page/create-new-skill.modal.controller.ts

```diff
@@ -0,0 +1,97 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+/**
+ * @fileoverview Controller for CreateNewSkillModal.
+ */
+
+require('domain/utilities/url-interpolation.service.ts');
+
+angular.module('oppia').controller('CreateNewSkillModal', [
+  '$scope', '$uibModalInstance', 'RubricObjectFactory', 'SkillCreationService',
+  'SkillObjectFactory', 'MAX_CHARS_IN_SKILL_DESCRIPTION',
+  'SKILL_DESCRIPTION_STATUS_VALUES', 'SKILL_DIFFICULTIES',
+  function($scope, $uibModalInstance, RubricObjectFactory, SkillCreationService,
+      SkillObjectFactory, MAX_CHARS_IN_SKILL_DESCRIPTION,
+      SKILL_DESCRIPTION_STATUS_VALUES, SKILL_DIFFICULTIES) {
+    var rubrics = [
+      RubricObjectFactory.create(SKILL_DIFFICULTIES[0], []),
+      RubricObjectFactory.create(SKILL_DIFFICULTIES[1], ['']),
+      RubricObjectFactory.create(SKILL_DIFFICULTIES[2], [])];
+    $scope.newSkillDescription = '';
+    $scope.rubrics = rubrics;
+    $scope.errorMsg = '';
+    $scope.bindableDict = {
+      displayedConceptCardExplanation: ''
+    };
+    $scope.MAX_CHARS_IN_SKILL_DESCRIPTION = (
+      MAX_CHARS_IN_SKILL_DESCRIPTION);
+    var newExplanationObject = null;
+
+    $scope.$watch('newSkillDescription', function() {
+      if (
+        SkillCreationService.getSkillDescriptionStatus() !==
+                SKILL_DESCRIPTION_STATUS_VALUES.STATUS_DISABLED) {
+        var initParagraph = document.createElement('p');
+        var explanations = $scope.rubrics[1].getExplanations();
+        var newExplanation = document.createTextNode(
+          $scope.newSkillDescription);
+        initParagraph.appendChild(newExplanation);
+        explanations[0] = initParagraph.outerHTML;
+        $scope.rubrics[1].setExplanations(explanations);
+        SkillCreationService.markChangeInSkillDescription();
+      }
+    });
+
+    $scope.onSaveExplanation = function(explanationObject) {
+      newExplanationObject = explanationObject.toBackendDict();
+      $scope.bindableDict.displayedConceptCardExplanation = (
+        explanationObject.getHtml());
+    };
+
+    $scope.onSaveRubric = function(difficulty, explanations) {
+      for (var idx in $scope.rubrics) {
+        if ($scope.rubrics[idx].getDifficulty() === difficulty) {
+          $scope.rubrics[idx].setExplanations(explanations);
+        }
+      }
+    };
+
+    $scope.resetErrorMsg = function() {
+      $scope.errorMsg = '';
+    };
+
+    $scope.createNewSkill = function() {
+      if (
+        !SkillObjectFactory.hasValidDescription(
+          $scope.newSkillDescription)) {
+        $scope.errorMsg = (
+          'Please use a non-empty description consisting of ' +
+                    'alphanumeric characters, spaces and/or hyphens.');
```

Comment:
Indentation is wrong. /cc @Hudda

Reply:
@seanlip Already in #8423

Reply:
(Just copy pasted this bit, hence missed that). Fixed.